src/rocm_docs/doxygen.py

```diff
@@ -93,19 +93,27 @@ def _run_doxygen(app: Sphinx, config: Config) -> None:
             " and be readable."
         ) from err
 
+    if config["doxygen_executable"] is None:
+        doxygen_exe = shutil.which("doxygen")
+    else:
+        doxygen_path: Path = Path(config["doxygen_executable"]).absolute()
+        if doxygen_path.is_file():
+            doxygen_exe = str(doxygen_path)
+        else:
+            doxygen_exe = None
+    if doxygen_exe is None:
+        raise RuntimeError(
+            "'doxygen' command not found! Make sure that "
+            "doxygen is installed and in the PATH or specify "
+            "via doxygen_executable configuration variable."
+        )
```

Comment:
```suggestion
    if config["doxygen_executable"] is None:
        cmd_path = shutil.which("doxygen")
        if cmd_path is None:
           raise RuntimeError(
                "'doxygen' command not found! Make sure that "
                "doxygen is installed and in the PATH or specify "
                "via doxygen_executable configuration variable."
            )
        doxygen_exe = Path(cmd_path)
    else:
        doxygen_exe = Path(app.confdir, config.doxygen_executable)
```

1. The type is inferred in the case of `variable = Path(...)` so specifying it explicitly is redundant (ie. `variable: Path = Path(...)` is not needed.
2. I don't think you should check the user provided config value and doing it halfway (i.e. it could be a file, but not be exectuable etc), just leave that to subprocess. ("It is better to ask for forgiveness rather than permission", i.e. avoiding races)
3. `config.key` is generally seems to be preferred in this file over `config['key']`